bigbluebutton-html5/client/main.jsx

```diff
@@ -2,18 +2,33 @@
 import React from 'react';
 import { Meteor } from 'meteor/meteor';
 import { render } from 'react-dom';
-import renderRoutes from '/imports/startup/client/routes';
 import logger from '/imports/startup/client/logger';
+import LoadingScreen from '/imports/ui/components/loading-screen/component';
+import { joinRouteHandler, authenticatedRouteHandler } from '/imports/startup/client/auth';
+import Base from '/imports/startup/client/base';
+import { Session } from 'meteor/session';
 
 Meteor.startup(() => {
-  render(renderRoutes(), document.getElementById('app'));
+  render(<LoadingScreen />, document.getElementById('app'));
+
   // Logs all uncaught exceptions to the client logger
   window.addEventListener('error', (e) => {
-    const stack = e.error.stack;
+    const { stack } = e.error;
     let message = e.error.toString();
 
     // Checks if stack includes the message, if not add the two together.
     (stack.includes(message)) ? message = stack : message += `\n${stack}`;
     logger.error(message);
   });
+
+  // TODO make this a Promise
+  joinRouteHandler((value, error) => {
+    authenticatedRouteHandler((valueInner, errorInner) => {
```

Comment:
Ideally both `joinRouteHandler` and `authenticatedRouteHandler` would be a components and renderer inside <Base>

Reply:
Updated the TODO list in the description - from my point of view this will be better off in a separate issue created as this pull request gets merged